ssvsigner/web3signer/web3signer.go

```
@@ -0,0 +1,215 @@
+	logger.Info("importing keystores")
+
+	payload := ImportKeystoreRequest{
+		Keystores:          keystoreList,
+		Passwords:          keystorePasswordList,
+		SlashingProtection: "", // TODO
+	}
+
+	body, err := json.Marshal(payload)
+	if err != nil {
+		logger.Error("failed to marshal payload", zap.Error(err))
+		return nil, fmt.Errorf("marshal payload: %w", err)
+	}
+
+	url := fmt.Sprintf("%s/eth/v1/keystores", c.baseURL)
+	req, err := http.NewRequest(http.MethodPost, url, bytes.NewReader(body))
+	if err != nil {
+		logger.Error("failed to create http request", zap.Error(err))
+		return nil, fmt.Errorf("create request: %w", err)
+	}
+	req.Header.Set("Content-Type", "application/json")
+
+	httpResp, err := c.httpClient.Do(req)
+	if err != nil {
+		logger.Error("failed to send http request", zap.Error(err))
+		return nil, fmt.Errorf("request failed: %w", err)
+	}
+	defer func() {
+		if err := httpResp.Body.Close(); err != nil {
+			logger.Error("failed to close http response body", zap.Error(err))
+		}
+	}()
+
+	respBytes, err := io.ReadAll(httpResp.Body)
+	if err != nil {
+		logger.Error("failed to read http response body", zap.Error(err))
+		return nil, fmt.Errorf("read response body: %w", err)
+	}
+	var resp ImportKeystoreResponse
+	if err := json.Unmarshal(respBytes, &resp); err != nil {
+		logger.Error("failed to unmarshal http response body", zap.String("body", string(respBytes)), zap.Error(err))
+		return nil, fmt.Errorf("unmarshal response: %w", err)
+	}
+
+	if httpResp.StatusCode != http.StatusOK {
+		logger.Error("failed to import keystore",
+			zap.Int("status_code", httpResp.StatusCode),
+			zap.String("message", resp.Message))
+		return nil, fmt.Errorf("unexpected status %d: %v", httpResp.StatusCode, resp.Message)
+	}
+
+	logger.Info("imported keystores", zap.Any("response", string(respBytes)))
+
+	var statuses []Status
+	for _, data := range resp.Data {
+		statuses = append(statuses, data.Status)
+	}
+
+	return statuses, nil
+}
```

**Issue: delayed handling of keystore import errors**

Currently, if `Web3Signer` returns `HTTP 200 OK` but contains an **error status inside `resp.Data`**, we do not handle it immediately. Instead, the caller must check the returned statuses and handle errors accordingly.

Why this is a problem?
1. **Errors are propagated to the caller instead of being handled at the earliest opportunity.**
2. **The function gives a false impression of success** (since it returns `HTTP 200 OK`), even though the actual import may have failed.
3. **Callers may need to add redundant error checks** to ensure they handle failures correctly.

**Suggested fix:**
Instead of returning all statuses to the caller, **handle `"error"` statuses immediately** and return an error from `ImportKeystore`. Example:

```go
var statuses []Status
for _, data := range resp.Data {
    if data.Status == StatusError {
        logger.Error("keystore import failed", zap.String("message", data.Message))
        return nil, fmt.Errorf("keystore import failed: %s", data.Message)
    }
    statuses = append(statuses, data.Status)
}
```

Also the **keystore import completed** logging could be improved to log only relevant details (status + publicKeys)

---

`web3signer` is designed to handle multiple keys in one request, so its API is designed keeping this in mind, and I think its client must support handling multiple keys even if we use just one. So if we return an error as you suggest, we won't add the rest so it will be inefficient.

I think we need to return an error if we receive `StatusError` in the ssv-signer client though. Fixed